nomad/leader.go

```diff
@@ -774,16 +774,47 @@ func (s *Server) restorePeriodicDispatcher() error {
 			continue
 		}
 
+		// We skip if the job doesn't allow overlap and there are already
+		// instances running
+		needed, err := s.isNewEvalNeeded(job)
+		if err != nil {
+			return fmt.Errorf("failed to get job status: %v", err)
+		}
+
```

Comment:
nit: removing this line makes it more apparent that the check is linked to the previous call I feel
```suggestion
```